tmail-backend/event-bus-postgres/src/main/java/org/apache/james/events/TMailEventDispatcher.java

```diff
@@ -0,0 +1,223 @@
+package org.apache.james.events;
+
+import static com.rabbitmq.client.MessageProperties.PERSISTENT_TEXT_PLAIN;
+import static org.apache.james.backends.rabbitmq.Constants.AUTO_DELETE;
+import static org.apache.james.backends.rabbitmq.Constants.DIRECT_EXCHANGE;
+import static org.apache.james.backends.rabbitmq.Constants.DURABLE;
+import static org.apache.james.backends.rabbitmq.Constants.EMPTY_ROUTING_KEY;
+import static org.apache.james.backends.rabbitmq.Constants.EXCLUSIVE;
+import static org.apache.james.backends.rabbitmq.QueueArguments.NO_ARGUMENTS;
+import static org.apache.james.events.KeyChannelMessage.quoteChannel;
+import static org.apache.james.events.PostgresEventBusModule.PostgresEventBusNotificationBindingsTable.CHANNEL;
+import static org.apache.james.events.PostgresEventBusModule.PostgresEventBusNotificationBindingsTable.ROUTING_KEY;
+import static org.apache.james.events.PostgresEventBusModule.PostgresEventBusNotificationBindingsTable.TABLE_NAME;
+import static org.apache.james.events.RabbitMQAndPostgresEventBus.EVENT_BUS_ID;
+
+import java.time.Duration;
+import java.util.Collection;
+import java.util.Optional;
+import java.util.Set;
+
+import org.apache.james.backends.postgres.utils.PostgresExecutor;
+import org.apache.james.backends.rabbitmq.RabbitMQConfiguration;
+import org.apache.james.events.RoutingKeyConverter.RoutingKey;
+import org.apache.james.util.MDCBuilder;
+import org.apache.james.util.MDCStructuredLogger;
+import org.apache.james.util.StructuredLogger;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import com.google.common.collect.ImmutableList;
+import com.google.common.collect.ImmutableMap;
+import com.google.common.collect.ImmutableSet;
+import com.rabbitmq.client.AMQP;
+
+import reactor.core.publisher.Flux;
+import reactor.core.publisher.Mono;
+import reactor.core.scheduler.Schedulers;
+import reactor.rabbitmq.BindingSpecification;
+import reactor.rabbitmq.ExchangeSpecification;
+import reactor.rabbitmq.OutboundMessage;
+import reactor.rabbitmq.Sender;
+import reactor.util.function.Tuples;
+import reactor.util.retry.Retry;
+
+public class TMailEventDispatcher {
```

Comment:
Totally understandable for the POC, but maybe we should try to refactor and generify something here with the existing TMailEventDispatcher with redis if we want to merge this work?

Not sure how difficult would that be though